A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js

```diff
@@ -309,12 +308,16 @@ class CDUPerformancePage {
                 }
             }
             mcdu.onRightInput[3] = (value) => {
-                if (value === "") {
-                    mcdu._toFlexChecked = true;
-                } else if (mcdu.setPerfTOFlexTemp(value)) {
-                    mcdu.tryCheckToData();
+                if (mcdu._toFlexChecked) {
+                    if (mcdu.setPerfTOFlexTemp(value)) {
+                        CDUPerformancePage.ShowTAKEOFFPage(mcdu);
+                    }
+                } else {
+                    if (value === "" || mcdu.setPerfTOFlexTemp(value)) {
+                        mcdu._toFlexChecked = true;
+                        CDUPerformancePage.ShowTAKEOFFPage(mcdu);
+                    }
```

Comment:
I see why you're doing this, however it's very confusing to read because it looks so redundant.

Reply:
I could refactor this to something like this:

```js
if (mcdu._toFlexChecked) {
   mcdu.onRight[3] = (value) => {
      if (mcdu.setPerfTOFlexTemp(value)) {
         CDUPerformancePage.ShowTAKEOFFPage(mcdu);
      }
   };
} else {
   mcdu.onRight[3] = (value) => {
      if (value === "" || mcdu.setPerfTOFlexTemp(value)) {
          mcdu._toFlexChecked = true;
          CDUPerformancePage.ShowTAKEOFFPage(mcdu);
      }
   };
}
```

Reply:
Or like this:
```js
mcdu.onRight[3] = (value) => {
   if (mcdu._toFlexChecked) {
      mcdu.setPerfTOFlexTemp(value);
   } else {
      if (value === "" || mcdu.setPerfTOFlexTemp(value)) {
          mcdu._toFlexChecked = true;
      }
   }
   CDUPerformancePage.ShowTAKEOFFPage(mcdu);
};
```